packages/eslint-plugin/tests/docs.test.ts

```diff
@@ -180,6 +181,32 @@ describe('Validating rule docs', () => {
           }
         });
       }
+
+      test('must include only valid code samples', () => {
+        for (const token of tokens) {
+          if (token.type !== 'code') {
+            continue;
+          }
+
+          const lang = token.lang?.trim();
+          if (!lang?.startsWith('ts')) {
+            return;
+          }
+
+          try {
+            parseForESLint(token.text, {
+              ecmaFeatures: {
+                jsx: lang.startsWith('tsx'),
```

Comment:
I still think that instead of `startsWith` you should use regex. Unless you plan to support things like `tsx-noformat` (a trick we use in MDN to stop Prettier from formatting code). This way we can be future proof and avoid some confusing behaviors.

Reply:
Changed it